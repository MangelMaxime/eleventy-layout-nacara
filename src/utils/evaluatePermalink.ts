
export const evaluatePermalink = (ctx: any) => {
    let permalink = null;
    if (ctx.data.permalink) {
        if (typeof ctx.data.permalink === "function") {
            permalink = ctx.data.permalink(ctx.data)
        } else {
            permalink = ctx.data.permalink;
        }
    }
    return permalink;
}
