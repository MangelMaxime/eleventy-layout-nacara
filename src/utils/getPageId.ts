import path from "path";
// import NacaraTypes from "./../../types/nacara";

export default function getPageId(fileStem : string) {
    //  Normal the path, so we can split using the path separator
    const normalizedInputPath = path.normalize(fileStem);
    // Extract all the segments of the path
    const inputPathSegments = normalizedInputPath.split(path.sep);
    // console.log("inputPathSegments:", inputPathSegments);
    // Build the section direction, which consist of the root + the first segment of the path
    const pageIdSegments = inputPathSegments.slice(2);

    return pageIdSegments.join("/");
};
