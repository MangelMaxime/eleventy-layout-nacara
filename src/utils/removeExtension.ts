import path from "path";

export function removeExtension(fileName: string): string {
    const parsedPath = path.parse(fileName);
    return path.join(parsedPath.dir, parsedPath.name);
}
