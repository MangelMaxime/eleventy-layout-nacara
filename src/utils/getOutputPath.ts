import { evaluatePermalink } from "./evaluatePermalink.js";

export default function getOutputPath(pageContext: any) {
    const trimOutputRegex = new RegExp(`^${pageContext.data.eleventy.directories.output}`);
    const permalink = evaluatePermalink(pageContext);

    if (permalink) {
        return permalink;
    }

    return pageContext.data.page.outputPath.replace(trimOutputRegex, "");
}
