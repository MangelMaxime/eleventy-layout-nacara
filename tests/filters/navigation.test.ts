import { expect, test } from 'vitest'
import { formatHTML } from "../utils/_formatHTML";
const Eleventy = require("@11ty/eleventy");

test("returns nothing if there is not nacaraMenu provided", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-0/",
        "./fixtures/navigation-0/_site",
        {
            configPath: "./fixtures/navigation-0/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});

test("returns the Previous button marked as invisible and the second button normal if this is the first page", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-1/",
        "./fixtures/navigation-1/_site",
        {
            configPath: "./fixtures/navigation-1/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page1Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page1/"
    );
    const formattedResult = formatHTML(page1Json.content);

    expect(formattedResult).toMatchSnapshot();
});

test("returns the Next button marked as invisible and the first button normal if this is the last page", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-2/",
        "./fixtures/navigation-2/_site",
        {
            configPath: "./fixtures/navigation-2/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page2Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page2/"
    );
    const formattedResult = formatHTML(page2Json.content);

    expect(formattedResult).toMatchSnapshot();
});

test("the Next button include the section title page is inside a section", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-3/",
        "./fixtures/navigation-3/_site",
        {
            configPath: "./fixtures/navigation-3/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page2Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page2/"
    );
    const formattedResult = formatHTML(page2Json.content);

    expect(formattedResult).toMatchSnapshot();
});

test("the Previous button include the section title page is inside a section", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-4/",
        "./fixtures/navigation-4/_site",
        {
            configPath: "./fixtures/navigation-4/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page2Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page1/"
    );
    const formattedResult = formatHTML(page2Json.content);

    expect(formattedResult).toMatchSnapshot();
});

test("both buttons are present if the page is between 2 valid element in the menu", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-5/",
        "./fixtures/navigation-5/_site",
        {
            configPath: "./fixtures/navigation-5/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page2Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page2/"
    );
    const formattedResult = formatHTML(page2Json.content);

    expect(formattedResult).toMatchSnapshot();
});

test("the Next button is empty if the next menu element is a link", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-6/",
        "./fixtures/navigation-6/_site",
        {
            configPath: "./fixtures/navigation-6/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page2Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page2/"
    );

    const formattedResult = formatHTML(page2Json.content);

    expect(formattedResult).toMatchSnapshot();
});

test("the Previous button is empty if the previous menu element is a link", async () => {
    const elev = new Eleventy(
        "./fixtures/navigation-7/",
        "./fixtures/navigation-7/_site",
        {
            configPath: "./fixtures/navigation-7/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const page2Json = json.find(
        (item: any) => item.url === "/docs/getting-started/page2/"
    );
    const formattedResult = formatHTML(page2Json.content);

    expect(formattedResult).toMatchSnapshot();
});
