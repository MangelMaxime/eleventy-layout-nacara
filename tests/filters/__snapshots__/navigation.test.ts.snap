// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`both buttons are present if the page is between 2 valid element in the menu 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-prev"
        href="/Nacara//docs/getting-started/page1/"
    >
        <i>←</i>
        <span class="is-hidden-mobile">
            <em>Getting Started</em>
            <strong>Page 1</strong>
        </span>
    </a>
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-next"
        href="/Nacara//docs/getting-started/page3/"
    >
        <span class="is-hidden-mobile">
            <em>Getting Started</em>
            <strong>Page 3</strong>
        </span>
        <i>→</i>
    </a>
</div>
"
`;

exports[`returns nothing if there is not nacaraMenu provided 1`] = `""`;

exports[`returns the Next button marked as invisible and the first button normal if this is the last page 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-prev"
        href="/Nacara//docs/getting-started/page1/"
    >
        <i>←</i>
        <span class="is-hidden-mobile">
            <em></em>
            <strong>Page 1</strong>
        </span>
    </a>
    <a class="button navigate-to-next is-invisible"></a>
</div>
"
`;

exports[`returns the Previous button marked as invisible and the second button normal if this is the first page 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a class="button navigate-to-previous is-invisible"></a>
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-next"
        href="/Nacara//docs/getting-started/page2/"
    >
        <span class="is-hidden-mobile">
            <em></em>
            <strong>Page 2</strong>
        </span>
        <i>→</i>
    </a>
</div>
"
`;

exports[`the Next button include the section title page is inside a section 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-prev"
        href="/Nacara//docs/getting-started/page1/"
    >
        <i>←</i>
        <span class="is-hidden-mobile">
            <em>Getting Started</em>
            <strong>Page 1</strong>
        </span>
    </a>
    <a class="button navigate-to-next is-invisible"></a>
</div>
"
`;

exports[`the Next button is empty if the next menu element is a link 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-prev"
        href="/Nacara//docs/getting-started/page1/"
    >
        <i>←</i>
        <span class="is-hidden-mobile">
            <em>Getting Started</em>
            <strong>Page 1</strong>
        </span>
    </a>
    <a class="button navigate-to-next is-invisible"></a>
</div>
"
`;

exports[`the Previous button include the section title page is inside a section 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a class="button navigate-to-previous is-invisible"></a>
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-next"
        href="/Nacara//docs/getting-started/page2/"
    >
        <span class="is-hidden-mobile">
            <em>Getting Started</em>
            <strong>Page 2</strong>
        </span>
        <i>→</i>
    </a>
</div>
"
`;

exports[`the Previous button is empty if the previous menu element is a link 1`] = `
"<div class="section bd-docs-pagination bd-pagination-links">
    <a class="button navigate-to-previous is-invisible"></a>
    <a
        class="button bd-fat-button is-primary is-light bd-pagination-next"
        href="/Nacara//docs/getting-started/page3/"
    >
        <span class="is-hidden-mobile">
            <em>Getting Started</em>
            <strong>Page 3</strong>
        </span>
        <i>→</i>
    </a>
</div>
"
`;
